File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function () {};

DnaTranscriber.prototype.toRna = function (s) {
    return s.split("").map(replace).join("")
};
function replace(c) {
  switch (c){
      case 'G': return 'C';
      case 'C': return 'G';
      case 'T': return 'A';
      case 'A': return 'U';
      default: throw new Error("Invalid input")
  }
}

module.exports = DnaTranscriber;

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    result = ""
    map = {
        'A' : 'U',
        'G' : 'C',
        'C' : 'G',
        'T' : 'A',
    }
    for x in dna_strand:
        if x not in "ACGT":
            raise ValueError("Invalid DNA strand input")
        result += map[x]
    return result

